src/Amalgam/interpreter/InterpreterOpcodesBase.cpp

```diff
@@ -592,18 +597,39 @@ EvaluableNodeReference Interpreter::InterpretNode_ENT_WHILE(EvaluableNode *en)
 		if(AreExecutionResourcesExhausted())
 			return EvaluableNodeReference::Null();
 
+		SetTopCurrentIndexInConstructionStack(static_cast<double>(loop_iteration++));
+		SetTopPreviousResultInConstructionStack(previous_result);
+
 		//run each step within the loop
-		for(size_t i = 1; i < ocn.size(); i++)
+		EvaluableNodeReference new_result;
+		for(size_t i = 1; i < ocn_size; i++)
 		{
-			if(result != nullptr && result->GetType() == ENT_CONCLUDE)
-				return RemoveConcludeFromConclusion(result, evaluableNodeManager);
+			new_result = InterpretNode(ocn[i]);
+
+			if(new_result != nullptr && new_result->GetType() == ENT_CONCLUDE)
+			{
+				//if previous result is unconsumed, free if possible
+				previous_result = GetAndClearPreviousResultInConstructionStack(0);
```

Comment:
something's going on in this method that prevents previous_result from being assigned to a variable, only allowing direct access to it